File: backend/app/schemas/social.py
"""Pydantic Schemas for Social Accounts and OAuth flows.

Strict security guarantee: No schema exposes provider access tokens,
refresh tokens, or client secrets to the client.
"""
from datetime import datetime
from typing import Optional
import uuid

from pydantic import BaseModel, ConfigDict, Field

from app.models.enums import Platform, SocialAccountStatus


class SocialAccountRead(BaseModel):
    """Publicly safe representation of a connected social account.
    
    NEVER includes access_token, refresh_token, or cryptographic keys.
    """
    id: uuid.UUID
    company_id: uuid.UUID
    platform: Platform
    account_name: str
    platform_account_id: str
    status: SocialAccountStatus
    token_expires_at: Optional[datetime] = None
    last_connected_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class OAuthStartResponse(BaseModel):
    """Response returned when initiating an OAuth connection flow."""
    authorization_url: str = Field(..., description="External platform OAuth authorization redirect URL")
    state: str = Field(..., description="High-entropy CSRF state token bound to request")
    platform: Platform


class OAuthCallbackResponse(BaseModel):
    """Response returned upon successful OAuth callback completion."""
    message: str = "Social account connected successfully"
    account: SocialAccountRead


class TokenRefreshResponse(BaseModel):
    """Response returned upon successful manual or automated token refresh."""
    message: str = "Social account token refreshed successfully"
    account: SocialAccountRead

File: frontend/src/types/social.ts
/**
 * SocialOS Social Accounts & OAuth TypeScript Definitions.
 * 
 * Strict security invariant: No token or credential fields exist in these types.
 */

export type Platform = "INSTAGRAM" | "FACEBOOK" | "LINKEDIN" | "YOUTUBE";

export type SocialAccountStatus = "ACTIVE" | "DISCONNECTED" | "EXPIRED" | "ERROR";

export interface SocialAccount {
  id: string;
  company_id: string;
  platform: Platform;
  account_name: string;
  platform_account_id: string;
  status: SocialAccountStatus;
  token_expires_at: string | null;
  last_connected_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface OAuthStartResponse {
  authorization_url: string;
  state: string;
  platform: Platform;
}

export interface OAuthCallbackResponse {
  message: string;
  account: SocialAccount;
}

export interface TokenRefreshResponse {
  message: string;
  account: SocialAccount;
}

export interface SocialProviderInfo {
  platform: Platform;
  name: string;
  description: string;
  iconBg: string;
  iconColor: string;
  supportedFeatures: string[];
}
